src/devicestate/ipcinterface.cpp

```diff
@@ -58,7 +58,15 @@ QList<QVariant> IPCInterface::get(const QString& method,
                                     const QVariant& arg1,
                                     const QVariant& arg2) {
     QList<QVariant> results;
-    QDBusMessage msg = call(method, arg1, arg2);
+
+    QDBusMessage msg;
+    if (arg1.isValid() && arg2.isValid()) {
+        msg = call(method, arg1, arg2);
+    } else if (arg1.isValid()) {
+        msg = call(method, arg1);
+    } else {
+        msg = call(method);
+    }
```

Comment:
micronit: the most likely (or the only?) use -> repeated arg1 checks -  could be avoided by e.g.

```
if (!arg1.isValid()) {
    msg = call(method);
} else if (!arg2.isValid()) {
    msg = call(method, arg1);
} else {
    msg = call(method, arg1, arg2);
}
```